pkg/apis/discovery/validation/validation.go

```diff
@@ -173,3 +178,33 @@ func validateAddressType(addressType discovery.AddressType) field.ErrorList {
 
 	return allErrs
 }
+
+func validateHints(endpointHints *discovery.EndpointHints, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if endpointHints == nil {
```

Comment:
this is duplicate, you are already checking that this is not nil before calling validateHints()

Reply:
Yeah good call, that was intentional but likely overkill, felt weird not to have a nil check but also unlikely that anything else will ever call this. Removed.

Reply:
sounds good, disregard the comment then please